table/block_based/block_prefetcher.cc

```diff
@@ -16,17 +16,31 @@ void BlockPrefetcher::PrefetchIfNeeded(const BlockBasedTable::Rep* rep,
   if (is_for_compaction) {
     rep->CreateFilePrefetchBufferIfNotExists(compaction_readahead_size_,
                                              compaction_readahead_size_,
-                                             &prefetch_buffer_);
+                                             &prefetch_buffer_, false);
     return;
   }
 
   // Explicit user requested readahead
   if (readahead_size > 0) {
     rep->CreateFilePrefetchBufferIfNotExists(readahead_size, readahead_size,
-                                             &prefetch_buffer_);
+                                             &prefetch_buffer_, false);
     return;
   }
 
+  // Enable implicit auto readahead if reads are sequential based on prev
+  // offset/length read and num_file_reads_.
+  if (prev_len_ != 0 && (prev_offset_ > handle.offset() ||
+                         prev_offset_ + prev_len_ < handle.offset())) {
+    prev_offset_ = handle.offset();
+    prev_len_ = block_size(handle);
```

Comment:
Do we need ```prev_len_```? Can ```prev_offset_ = handle.offset() + block_size(handle)```?

Reply:
In order to skip first block checking I added prev_len_. It checks ```if (prev_len_!=0 && not sequential condition)```.